.github/workflows/on-push.yml

```diff
@@ -0,0 +1,114 @@
+name: on-push
+
+on:
+  push:
+    branches:
+    - main
+    tags:
+    - "*"
+  pull_request:
+    branches:
+    - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install the latest version of uv
+      uses: astral-sh/setup-uv@v5
+      with:
+        version: "latest"
+    - uses: pre-commit/action@v3.0.1
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12", "3.13"] # Add more versions as needed.
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install the latest version of uv and set the python version
+      uses: astral-sh/setup-uv@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - uses: extractions/setup-just@v2
+    - name: Test with python ${{ matrix.python-version }}
+      run: just unit-tests
+
+  type-check:
+    needs: [unit-tests]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install the latest version of uv
+      uses: astral-sh/setup-uv@v5
+      with:
+        version: "latest"
+    - name: Install type-checking dependencies
+      run: uv sync --group type-checking
+    - uses: extractions/setup-just@v2
+    - name: Run code quality checks
+      run: just type-check
+
+  docs-build:
+    # needs: [unit-tests]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install the latest version of uv and set the python version to 3.13
+      uses: astral-sh/setup-uv@v5
+      with:
+        version: "latest"
+    # - name: Install documentation dependencies
+      # run: uv sync --group docs
+    - name: Install sphinx  
+      run: |
+        uv --systeù pip install .
```

Comment:
uv will install the current package as part of `sync` by default, so I think you only need the `uv sync --group docs`.